src/components/ThoughtAnnotation.js

```diff
@@ -43,16 +43,21 @@ export const ThoughtAnnotation = connect(({ cursor, cursorBeforeEdit, focusOffse
     dark: !meta([EM_TOKEN, 'Settings', 'Theme']).Light,
     thoughtsRanked: thoughtsRankedLive,
     isEditing,
-    focusOffset
+    focusOffset,
+    editingValue,
+    invalidState
   }
-})(({ dark, thoughtsRanked, showContexts, showContextBreadcrumbs, contextChain, homeContext, isEditing, focusOffset, minContexts = 2, url, dispatch }) => {
+})(({ dark, thoughtsRanked, showContexts, showContextBreadcrumbs, contextChain, homeContext, isEditing, focusOffset, minContexts = 2, url, dispatch, invalidState, editingValue }) => {
 
   // disable intrathought linking until add, edit, delete, and expansion can be implemented
   // get all subthoughts and the subthought under the selection
+
+  const isRealTimeContextUpdate = isEditing && invalidState && editingValue !== null // only show real time update if being edited while having meta validation error
```

Comment:
> I'm not sure I understand the reason for `&& invalidState`. Don't we want to update the `contexts` regardless of whether it is in a valid or invalid state?

When thought is in a `invalid-state` we know that `exitingThoughtChange` is not called. So the `numContext` of the `editingValue` should be increased by 1. But in a valid state `exitingThoughtChange` is called with  `newValue` and `numContext` is already in sync for currently edited thought. If we don't use `invalidState` flag , superscript will always be increased by 1 for edited thought.

Reply:
Okay, that makes sense. Maybe add a comment like: 

`// do not increase numContexts when in an invalid state since the thought has not been updated in state`